### PR TITLE
feat: Migrate to eslint from tslint

### PR DESCRIPTION
**Pull request by shawn-dsz**
## Purpose

Be in-line with wider Typescript Community - this means migrating deprecating TSLint in favour of ESLint. 

This includes BREAKING CHANGE: from [`eslint-config-seek`](https://github.com/seek-oss/eslint-config-seek/pull/36) hence you may need to update your codebase to be in line with SEEK's lint recommendations.

### CODE REVIEW

File: lib/runESLint.js [hunk 1]
```diff
@@ -29,17 +30,21 @@ const runESLint = ({ fix = false, paths }) =>
...
-        const { errorCount, warningCount, results } = cli.executeOnFiles(
-          filteredFilePaths,
-        );
+        const report = cli.executeOnFiles(filteredFilePaths);
 
-        if (errorCount || warningCount) {
-          const formatter = cli.getFormatter();
-          console.log(formatter(results));
-        }
+        if (fix) {
```
Comment (mattsjones) [inline lib/runESLint.js]: This change means when running in `fix` mode we don't exit with errors if something is not fixable.

File: test/test-cases/seek-style-guide/seek-style-guide.test.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable jest/expect-expect */
+/* eslint-disable jest/expect-expect */
```
Comment (mattsjones) [inline test/test-cases/seek-style-guide/seek-style-guide.test.js]: This is here twice

File: docs/docs/linting.md [hunk 3]
```diff
@@ -1,8 +1,8 @@
 # Linting and Formatting
 
-(via [ESLint](http://eslint.org/), [TSLint](https://palantir.github.io/tslint/) and [Prettier](https://github.com/prettier/prettier))
+(via [ESLint](http://eslint.org/) and [Prettier](https://github.com/prettier/prettier))
 
-Running `sku lint` will execute the ESLint/TSLint rules over the code in your `src` directory, depending on the type of file. You can see the ESLint rules defined for sku projects in [eslint-config-seek](https://github.com/seek-oss/eslint-config-seek). Similarly you can see the TSLint rules defined in [tslint-config-seek](https://github.com/seek-oss/tslint-config-seek).
+Running `sku lint` will execute the ESLint rules over the code in your `src` directory, depending on the type of file. You can see the ESLint rules defined for sku projects in [eslint-config-seek](https://github.com/seek-oss/eslint-config-seek). Support is provided for both Javascript and Typescript files.
```
Comment (michaeltaranto) [inline docs/docs/linting.md]:
```suggestion
Running `sku lint` will execute the ESLint rules over the code in your `src` directory. You can see the ESLint rules defined for sku projects in [eslint-config-seek](https://github.com/seek-oss/eslint-config-seek). Support is provided for both Javascript and Typescript files.
```

File: lib/runESLint.js [hunk 4]
```diff
@@ -8,17 +8,18 @@ const runESLint = ({ fix = false, paths }) =>
 
     const cli = new EslintCLI({
       baseConfig: eslintConfig,
+      extensions: ['.ts', '.tsx', '.js', '.jsx'],
```
Comment (michaeltaranto) [inline lib/runESLint.js]: Should we add `json` as per the path filter below?

File: docs/docs/linting.md [hunk 5]
```diff
@@ -4,6 +4,6 @@
 
 Running `sku lint` will execute the ESLint rules over the code in your `src` directory, depending on the type of file. You can see the ESLint rules defined for sku projects in [eslint-config-seek](https://github.com/seek-oss/eslint-config-seek). Support is provided for both Javascript and Typescript files.
 
-Running `sku format` will format all JavaScript and TypeScript files in your project using [Prettier](https://github.com/prettier/prettier), and fix [ESLint](http://eslint.org/) errors where possible. As changes to formatting are considered non-breaking, please ensure you run `sku format` after upgrading `sku`.
+Running `sku lint` will execute the ESLint rules over the code in your `src` directory. You can see the ESLint rules defined for sku projects in [eslint-config-seek](https://github.com/seek-oss/eslint-config-seek). Support is provided for both Javascript and Typescript files.
```
Comment (michaeltaranto) [inline docs/docs/linting.md]: Looks like a funky merge here, we lost the `sku format` docs 🤔

File: lib/runESLint.js (hunk 4 above)
Comment (shawn-dsz) [reply lib/runESLint.js]: I did try that, but it's giving false positives on`json` files

**Pull request merged**